src/main/scala/info/bethard/timenorm/formal/Types.scala

```diff
@@ -233,7 +233,10 @@ case class Year(digits: Int, nMissingDigits: Int = 0) extends Interval {
   */
 case class YearSuffix(interval: Interval, lastDigits: Int, nMissingDigits: Int = 0) extends Interval {
   val isDefined: Boolean = interval.isDefined
-  val nSuffixDigits: Int = (math.log10(lastDigits) + 1).toInt
+  val nSuffixDigits: Int = lastDigits match {
```

Comment:
Please add a test for this.

Reply:
I have found that YearSuffix didn't work when a Two-Digit-Year had leading zeros. I have made some changes to deal with this and also added a test.